src/userDashboard/UserDashboard.js

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from "react";
+import LoadingAnimation from "../components/LoadingAnimation";
+import UserLineGraph from "./userGraphs/UserLineGraph";
+import UserBarGraph from "./userGraphs/UserBarGraph";
+import {PERIOD_OPTIONS, ACTIVITY_TIME_FORMAT_OPTIONS} from "./dataFormat/ConstantsUserDashboard";
+import { getLineGraphData, calculateCountPerMonth_Words, getMapData} from "./dataFormat/LineGraphDataFormat";
+import { getBarGraphData, calculateCountPerMonth_Activity } from "./dataFormat/BarGraphDataFormat";
+
+import UserDashboard_Top from "./UserDashboard_Top";
+
+export default function UserDashboard({ api }){
+
+    const [activeTab, setActiveTab] = useState(1);
+    const [activeOption, setActiveOption] = useState(PERIOD_OPTIONS.WEEK);
+    const [activeTimeFormatOption, setActiveTimeFormatOption] = useState(ACTIVITY_TIME_FORMAT_OPTIONS.MINUTES);
+    const [allWordsData, setAllWordsData] = useState(null);
+    const [allWordsDataPerMonths, setAllWordsDataPerMonths] = useState({});
+    const [dateForGraphs, setDateForGraphs] = useState(new Date());
+    const [userActivityData, setUserActivityData] = useState(null);
+    const [userActivityDataPerMonths, setUserActivityDataPerMonths] = useState({});

```

Comment:
why is this here?
it's suspicious that we need this perMonths but not a similar one for perYears. or?